src/clj_irods/core.clj

```diff
@@ -109,20 +119,23 @@
             listing-item (when (delay? paged-folder-listing)
                            (get-from-listing (apply concat (icat/flatten-cached-listings @paged-folder-listing))))]
         (or
-          (when listing-item (delay (extract-fn listing-item)))
-          (when single-item  (delay (extract-fn @single-item)))
+          (when listing-item (instrumented-delay (extract-fn listing-item)))
+          (when single-item  (instrumented-delay (extract-fn @single-item)))
           nil))
       ;; better if we pass user group IDs in so we can get them from any source, rather than forcing the icat version specifically
       (when (:has-icat irods)
-        (let [userids (icat/user-group-ids irods user zone)]
-          (delay (force userids) (extract-fn @(icat/get-item irods user zone path))))))))
+        (let [userids (icat/user-group-ids irods user zone)
+              item (future (force userids) @(icat/get-item irods user zone path))]
+          (instrumented-delay (extract-fn @item)))))))
 
 (defn- from-stat
   [cache? irods extract-fn path]
   (if cache?
     (let [jargon-stat (jargon/cached-stat irods path)]
-      (when jargon-stat (delay (extract-fn @jargon-stat))))
-    (when (:has-jargon irods) (delay (extract-fn @(jargon/stat irods path))))))
+      (when jargon-stat (instrumented-delay (extract-fn @jargon-stat))))
+    (when (:has-jargon irods)
+      (let [jargon-stat (jargon/stat irods path)]
+        (instrumented-delay (extract-fn @jargon-stat))))))
```

Comment:
This makes sure the jargon stat starts getting fetched right away, not when the delay is deref'd.